dojo/problem/redis.py

```diff
@@ -0,0 +1,155 @@
+import base64
+import json
+import logging
+from dataclasses import dataclass, field
+from functools import lru_cache
+
+import redis
+
+import dojo.problem.helper as problems_help
+from dojo.models import Finding
+
+logger = logging.getLogger(__name__)
+
+SEVERITY_ORDER = {
+    "Critical": 5,
+    "High": 4,
+    "Medium": 3,
+    "Low": 2,
+    "Info": 1,
+}
+
+
+@dataclass
+class Problem:
+    name: str
+    problem_id: str
+    severity: str
+    script_ids: set = field(default_factory=set)
+    finding_ids: set = field(default_factory=set)
+
+    def to_dict(self):
+        return {
+            "name": self.name,
+            "problem_id": self.problem_id,
+            "severity": self.severity,
+            "script_ids": list(self.script_ids),
+            "finding_ids": list(self.finding_ids),
+        }
+
+    @staticmethod
+    def from_dict(data):
+        return Problem(
+            name=data["name"],
+            problem_id=data["problem_id"],
+            severity=data["severity"],
+            script_ids=set(data.get("script_ids", [])),
+            finding_ids=set(data.get("finding_ids", [])),
+        )
+
+
+@lru_cache(maxsize=1)
+def get_redis_client():
+    return redis.Redis(host="django-defectdojo-redis-1", port=6379, decode_responses=True)
+
+
+def add_finding_to_redis(finding):
```

Comment:
This doesn't seem to be the correct name; it seems to be something like `update_finding_problem_on_redis`.  Maybe we should make it a staticmethod on class `ProblemHelper` or something.  So in other modules we would call `ProblemHelper.add_finding(finding)`.